go-controller/pkg/util/util.go

```diff
@@ -317,6 +318,37 @@ func IsClusterIP(svcVIP string) bool {
 	return false
 }
 
+// GetActiveNetworkForNamespace returns the active network for the given namespace
+// active network here means the network managing this namespace and responsible for
+// plumbing all the entities for this namespace
+// this is:
+// 1) "default" if there are no NADs in the namespace OR all NADs are primaryNetwork:false
+// 2) "<secondary-network-name>" if there is exactly ONE NAD with primaryNetwork:true
+// 3) "unknown" under all other conditions
+func GetActiveNetworkForNamespace(nads []*nadapi.NetworkAttachmentDefinition, podNamespace string) (string, error) {
```

Comment:
> The function should find the nads for a namespace so we don't open the door to users passing the wrong nads.

I can maybe pass that information in the func description? because its used from two places, podallocator that uses the lister directly and watchfactory that uses factory wrapper, so hard to pass the "find" nads...

I have a better idea, how about I make this `GetActiveNetworkForNamespaceFromNADs` ?

> 
> Also `podNamespace` should be renamed to just `namespace`

+1